drivers/counter/counter_esp32_rtc.c

```diff
@@ -235,9 +235,6 @@ static void counter_esp32_isr(void *arg)
 	if (data->alarm_cfg.callback) {
 		data->alarm_cfg.callback(dev, 0, now, data->alarm_cfg.user_data);
 	}
-
```

Comment:
`counter_esp32_cancel_alarm` at line 232 already clears interrupt. No need to re-do it here.